### PR TITLE
[docs]Refine Pulsar Functions overview and structure

### LINKED ISSUE #4554: [docs] Improve the structure of Pulsar Functions
**Is your feature request related to a problem? Please describe.**
I'm a little confused when searching information on Pulsar Functions. The content is scattered in different sections and does not integrate as a whole. Meanwhile, some information duplicates in *Overview* and other sections. It's a little bit frustrating to read the docs as a new user.

**Describe the solution you'd like**
I want to re-structure the architecture of Pulsar Functions as follows, so it will be easier to read and find information.
image

If you have any suggestion on the new architecture, feel free to contact me.
If there is no issue with it, I'll update the content throughout.

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Master Issue: #4554

### Motivation
We've proposed a new structure for Pulsar Functions in #4554, so I refine the overview file and part of structure.

### Modifications
1. Refine the Pulsar Functions overview.
2. Refine part of Pulsar Functions structure.
3. Currently, I keep the following parts in the sidebars.json unchanged.  We're designing the "Language-specific Code Tabs" for website. After success, we can adopt the structure as advised by @sijie in #4554, and I'll create new files for this part.
image

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@wolfstudy could you help review the update? Thank you.

**Comment by Jennifer88huang** [comment]
Good idea, I'll update it soon.

File: site2/docs/functions-overview-new.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+---
+id: functions-overview
+title: Pulsar Functions overview
+sidebar_label: Overview
+---
+
```
Comment (sijie) [inline site2/docs/functions-overview-new.md]: I would suggest adding a banner to indicate that this document is under construction.

```
> The section of the documentation is the new documentation of Pulsar Function. It is still under construction. If you would like to checkout the documentation of Pulsar Function. Please check out the old [Pulsar Functions](...) documentation.
```

Comment (Jennifer88huang) [reply site2/docs/functions-overview-new.md]: Done.

**Comment by wolfstudy** [comment]
In OverView, is it necessary to add `Function example` and `Deployment example` and so on.

I think

- We can move them to a specific API module to introduce
- Add the Function related to the `Function Worker` to OverView because

**Comment by Jennifer88huang** [comment]
> In OverView, is it necessary to add `Function example` and `Deployment example` and so on.
> * We can move them to a specific API module to introduce

We can move the two examples to specific API modules if that's more proper. @sijie any thoughts on this?

> * Add the Function related to the `Function Worker` to OverView because

We have put `Functions worker` in a separate section, refer to #4605.

**Comment by sijie** [comment]
@Jennifer88huang @wolfstudy  I think it might be good to keep an simple example in the overview section, so the developers get an idea about how does Pulsar Functions look like.

**Comment by Jennifer88huang** [comment]
Currently, we have four examples for programming model section: function, deployment, word count, content-based routing examples. So shall we keep the four examples, or just keep word count example?

**Comment by Jennifer88huang** [comment]
We've decided to keep "deploy example" and "word count example", and remove the other two examples.

**Comment by Jennifer88huang** [comment]
Updated as per decision. If there is no other issue, could we merge this PR?

**Comment by Jennifer88huang** [comment]
1. Since native functions/SDK, SerDe, Schema Registry, logging, user config, metrics, state storage have been written in [functions-deploy(new) guide](https://github.com/apache/pulsar/pull/4725), I remove those content from the "overview".
2. Add a new section for "thread/process mode", and fix https://github.com/apache/pulsar/issues/3400

**Comment by Jennifer88huang** [comment]
Any issue, feel free to comment. If there is no issue, could we merge this PR?

**Comment by Jennifer88huang** [comment]
run integration tests
run java8 tests

**Pull request merged**